langchain/schema.py

```diff
@@ -259,3 +259,32 @@ def clear(self) -> None:
 
 
 Memory = BaseMemory
+
+
+class BaseOutputParser(BaseModel, ABC):
+    """Class to parse the output of an LLM call."""
+
+    @abstractmethod
+    def parse(self, text: str) -> Any:
+        """Parse the output of an LLM call."""
+
+    def parse_with_prompt(self, completion: str, prompt: PromptValue) -> Any:
+        return self.parse(completion)
+
+    def get_format_instructions(self) -> str:
+        raise NotImplementedError
+
+    @property
+    def _type(self) -> str:
+        """Return the type key."""
+        raise NotImplementedError
+
+    def dict(self, **kwargs: Any) -> Dict:
+        """Return dictionary representation of output parser."""
+        output_parser_dict = super().dict()
+        output_parser_dict["_type"] = self._type
+        return output_parser_dict
+
+
+class OutputParserException(Exception):
```

Comment:
1 line doc-string to explain what the exception is to be used for and in what cases it's expected to be raised / excepted?

The exception name looks like it could be raised by either logic issues in the parser or due to malformed data. Is this exception meant to catch only the latter?

Reply:
added